src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -0,0 +1,557 @@
+	    when(this.order.getType()).thenReturn(InstanceType.VOLUME);
+	    when(volumePlugin.requestInstance(any(VolumeOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	@Test
+	public void testRequestNetworkInstance() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(NetworkOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.NETWORK);
+	    when(networkPlugin.requestInstance(any(NetworkOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
+		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
+	}
+	
+	// If plugin returns a null instance id, the method requestInstance() must throw an exception
+	@Test(expected = UnexpectedException.class)
+	public void testExceptionNullComputeInstanceId() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(ComputeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    when(computePlugin.requestInstance(any(ComputeOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	// If plugin returns a null instance id, the method requestInstance() must throw an exception
+	@Test(expected = UnexpectedException.class)
+	public void testExceptionNullNetworkInstanceId() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(NetworkOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.NETWORK);
+	    when(networkPlugin.requestInstance(any(NetworkOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	// If plugin returns a null instance id, the method requestInstance() must throw an exception
+	@Test(expected = UnexpectedException.class)
+	public void testExceptionNullAttachmentInstanceId() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(AttachmentOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.ATTACHMENT);
+	    when(attachmentPlugin.requestInstance(any(AttachmentOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	// If plugin returns a null instance id, the method requestInstance() must throw an exception
+	@Test(expected = UnexpectedException.class)
+	public void testExceptionNullVolumeInstanceId() throws FogbowManagerException, UnexpectedException {
+		this.order = Mockito.mock(VolumeOrder.class);
+	    when(this.order.getType()).thenReturn(InstanceType.VOLUME);
+	    when(volumePlugin.requestInstance(any(VolumeOrder.class), any(Token.class))).thenReturn(null);
+		this.localCloudConnector.requestInstance(order);
+	}
+	
+	// The order has an InstanceID, so the method getResourceInstance() is called.
+	@Test
+	public void testGetNetworkInstance() throws FogbowManagerException, UnexpectedException {
```

Comment:
where is the expected getResourceInstance behaviour configured, @jvmafra ?

Reply:
when(networkPlugin.getInstance(any(String.class), any(Token.class))).thenReturn(this.networkInstance);

getResourceInstance calls the method getInstance of plugin and we are mocking it.

Reply:
The documentation tries to say that the if branch will be executed because the order has an instance id